### PR TITLE
Refactor code for accessing database connections.

### PR DESCRIPTION
**Pull request by shradayshakya**
1. Refactor code for accessing database connections.
2. Update documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,4 @@
-sync-db
-=======
+# sync-db
 
 Database synchronization utility.
```
Comment (kabirbaidhya) [inline README.md]: Command line utility to synchronize and version control relational database objects across databases.

File: README.md [hunk 2]
```diff
@@ -9,11 +8,115 @@ Database synchronization utility.
 [![Language grade: JavaScript](https://img.shields.io/lgtm/grade/javascript/g/laudio/sync-db.svg?logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/laudio/sync-db/context:javascript)
 [![License](https://img.shields.io/npm/l/@leapfrogtechnology/sync-db.svg)](https://github.com/kabirbaidhya/sync-db/blob/master/package.json)
 
+# Set up
+
+## Install sync-db
+
+Install `sync-db` globally.
+
+    $ yarn global add @lftechnology/sync-db
```
Comment (kabirbaidhya) [inline README.md]:
```
npm install -g @leapfrogtechnology/sync-db
```

Comment (kabirbaidhya) [inline README.md]:
```
yarn global add @leapfrogtechnology/sync-db
```

File: README.md [hunk 3]
```diff
@@ -9,11 +8,115 @@ Database synchronization utility.
...
+Create `connections-sync-db.json` in your project and configure database connection(s) to be synced.
+
+```json
+{
+  "connections": [
+    {
+      "id": "db1",
+      "host": "localhost",
+      "port": 1433,
+      "user": "admin",
```
Comment (kabirbaidhya) [inline README.md]: db1

File: README.md [hunk 4]
```diff
@@ -9,11 +8,115 @@ Database synchronization utility.
...
+      "database": "foo",
```
Comment (kabirbaidhya) [inline README.md]: db1

File: README.md [hunk 5]
```diff
@@ -9,11 +8,115 @@ Database synchronization utility.
...
+      "password": "password",
+      "client": "mssql"
+    },
+    {
+      "id": "db2",
+      "host": "localhost",
+      "port": 1433,
+      "user": "admin",
```
Comment (kabirbaidhya) [inline README.md]: db2

File: README.md [hunk 6]
```diff
@@ -9,11 +8,115 @@ Database synchronization utility.
...
+      "database": "bar",
```
Comment (kabirbaidhya) [inline README.md]: db2

File: README.md [hunk 7]
```diff
@@ -9,11 +8,115 @@ Database synchronization utility.
...
+      "password": "password",
+      "client": "mysql2"
```
Comment (kabirbaidhya) [inline README.md]: We haven't tested this in mysql yet so let's keep it limited to `mssql` until we have another sample example for mysql.

File: README.md [hunk 8]
```diff
@@ -9,11 +8,115 @@ Database synchronization utility.
...
+
 # Usage
+
 ```bash
 $ sync-db
 ```
 
+# Sample Projects
+
+1. [Microsoft SQL Sample](examples/node-app-mssql)
```
Comment (kabirbaidhya) [inline README.md]:
```
Node MSSQL Sample (JavaScript)
```

File: examples/node-app-mssql/connections.sync-db.json.example [hunk 9]
```diff
@@ -1,11 +1,13 @@
...
-    "client": "mssql"
-  }
-]
+{
+  "connections": [
+    {
+      "id": "db1",
+      "host": "localhost",
+      "port": 1433,
+      "user": "admin",
```
Comment (kabirbaidhya) [inline examples/node-app-mssql/connections.sync-db.json.example]: db1

File: examples/node-app-mssql/connections.sync-db.json.example [hunk 10]
```diff
@@ -1,11 +1,13 @@
...
+      "database": "test",
```
Comment (kabirbaidhya) [inline examples/node-app-mssql/connections.sync-db.json.example]: db1

File: src/config.ts [hunk 11]
```diff
@@ -41,7 +41,7 @@ export async function resolveConnections(): Promise<Connection[]> {
   log('Resolving file: %s', filename);
 
   const loaded = await fs.read(filename);
-  const connections = JSON.parse(loaded);
+  const connections = (JSON.parse(loaded)).connections;
```
Comment (kabirbaidhya) [inline src/config.ts]:
```ts
const { connections } = JSON.parse(loaded);
```

File: README.md [hunk 12]
```diff
@@ -9,11 +8,115 @@ Database synchronization utility.
 [![Language grade: JavaScript](https://img.shields.io/lgtm/grade/javascript/g/laudio/sync-db.svg?logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/laudio/sync-db/context:javascript)
 [![License](https://img.shields.io/npm/l/@leapfrogtechnology/sync-db.svg)](https://github.com/kabirbaidhya/sync-db/blob/master/package.json)
 
+# Set up
+
+## Install sync-db
```
Comment (kabirbaidhya) [inline README.md]: Installation

File: README.md [hunk 13]
```diff
@@ -9,11 +8,115 @@ Database synchronization utility.
 [![Language grade: JavaScript](https://img.shields.io/lgtm/grade/javascript/g/laudio/sync-db.svg?logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/laudio/sync-db/context:javascript)
 [![License](https://img.shields.io/npm/l/@leapfrogtechnology/sync-db.svg)](https://github.com/kabirbaidhya/sync-db/blob/master/package.json)
 
+# Set up
```
Comment (kabirbaidhya) [inline README.md]: Setup

File: README.md [hunk 14]
```diff
@@ -1,27 +1,31 @@
...
-## Install sync-db
+## Installation of sync-db
 
 Install `sync-db` globally.
 
-    $ yarn global add @lftechnology/sync-db
+    $ yarn global add @leapfrogtechnology/sync-db
 
-Install database driver(s) of the database(s) that are to be synced.
+Globally, install node database driver(s) of the database(s) that are to be synced, for example this is how you would install driver mssql:
```
Comment (mesaugat) [inline README.md]: install the mssql driver.

Comment (mesaugat) [reply README.md]: Are we sure we want to make the end user install the package globally? Adding it to a existing project makes more sense.

Also, we have npx these days which makes global modules easier to run.

File: README.md (hunk 3 above)
Comment (mesaugat) [reply README.md]: db1user

File: README.md (hunk 5 above)
Comment (mesaugat) [reply README.md]: db2user

File: README.md [hunk 15]
```diff
@@ -1,19 +1,126 @@
...
+
+    $ yarn global add @leapfrogtechnology/sync-db
+
+Globally, install node database driver(s) of the database(s) that are to be synced, for example this is how you would install driver mssql:
+
+    $ yarn global add mssql
+
+## Configure Database Connections
+
+Create `connections-sync-db.json` in your project and configure database connection(s) to be synced.
```
Comment (mesaugat) [inline README.md]: your database connection(s).

File: README.md [hunk 16]
```diff
@@ -1,19 +1,126 @@
...
+
+Your file should look something like this
```
Comment (mesaugat) [inline README.md]: Remove

File: README.md [hunk 17]
```diff
@@ -1,19 +1,126 @@
...
+      "user": "admin",
+      "database": "db2",
+      "password": "password",
+      "client": "mssql"
+    }
+  ]
+}
+```
+
+## Configure Path to Path To SQL Database Objects
```
Comment (mesaugat) [inline README.md]: Path to Path?

File: README.md [hunk 18]
```diff
@@ -1,19 +1,126 @@
...
+
+1. Copy the sql files in your project in following folder structure
```
Comment (mesaugat) [inline README.md]: `SQL`

File: README.md [hunk 19]
```diff
@@ -1,19 +1,126 @@
...
+    │  │  ├─ procedure4.sql
+    │  │  └─ ...
+    │  ├─ procedure5.sql
+    │  └─ ...
+    │
+    └─...
+
+```
+
+**Note: When procedures and functions aren't placed inside a schema folder, they are associated with default schema.**
```
Comment (mesaugat) [inline README.md]: with the default schema.

File: README.md [hunk 20]
```diff
@@ -1,19 +1,126 @@
...
+
+```yml
+basePath: /path/to/sql
+sql:
+  - schema/<schema_name>.sql
+  - function/<schema_name>/<function_name>.sql
+  - procedure/<schema_name>/<procedure_name>.sql
+```
+
+**Note: Default the basePath would be `src/sql`.**
```
Comment (mesaugat) [inline README.md]: Move this to point number 2.

Comment (mesaugat) [reply README.md]: The default basePath is `src/sql`.

File: examples/node-app-mssql/connections.sync-db.json.example (hunk 9 above)
Comment (mesaugat) [reply examples/node-app-mssql/connections.sync-db.json.example]: db1user

File: README.md (hunk 14 above)
Comment (shradayshakya) [reply README.md]: Should I add documentation for both global and local alternatives?

File: README.md (hunk 1 above)
Comment (mesaugat) [reply README.md]: LGTM

File: README.md [hunk 21]
```diff
@@ -1,19 +1,122 @@
...
 
 [![oclif](https://img.shields.io/badge/cli-oclif-brightgreen.svg)](https://oclif.io)
 [![Version](https://img.shields.io/npm/v/@leapfrogtechnology/sync-db.svg)](https://npmjs.org/package/@leapfrogtechnology/sync-db)
 [![Downloads/week](https://img.shields.io/npm/dw/@leapfrogtechnology/sync-db.svg)](https://npmjs.org/package/@leapfrogtechnology/sync-db)
 [![Language grade: JavaScript](https://img.shields.io/lgtm/grade/javascript/g/laudio/sync-db.svg?logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/laudio/sync-db/context:javascript)
 [![License](https://img.shields.io/npm/l/@leapfrogtechnology/sync-db.svg)](https://github.com/kabirbaidhya/sync-db/blob/master/package.json)
 
+# Setup
+
+## Installation of sync-db
```
Comment (mesaugat) [inline README.md]: Only `Installation`

File: README.md [hunk 22]
```diff
@@ -1,19 +1,122 @@
...
+
+Install `sync-db` in your project.
```
Comment (mesaugat) [inline README.md]: Remove this as well because it is obvious.

File: README.md (hunk 18 above)
Comment (mesaugat) [reply README.md]: I mean capital `SQL`.

File: README.md [hunk 23]
```diff
@@ -1,19 +1,120 @@
...
+
+3. Specify the base path of the folder for SQL files through `basePath` key.
+
+4. List database objects in the order they need to be synced through `sql` parent.
+
 # Usage
+
 ```bash
-$ sync-db
+$ npx sync-db
```
Comment (mesaugat) [inline README.md]: .

Comment (mesaugat) [reply README.md]: Can we add this as a `npm script` instead?

Comment (mesaugat) [reply README.md]: Also, we can make another section in the readme where we can mention `npx`.

File: README.md [hunk 24]
```diff
@@ -1,19 +1,120 @@
...
+
+```yml
+basePath: /path/to/sql
+sql:
+  - schema/<schema_name>.sql
+  - function/<schema_name>/<function_name>.sql
+  - procedure/<schema_name>/<procedure_name>.sql
+```
+
+**Note: Default the basePath is `src/sql`.**
```
Comment (mesaugat) [inline README.md]:
```diff
- the
```

File: README.md [hunk 25]
```diff
@@ -99,16 +99,28 @@ sql:
...
 4. List database objects in the order they need to be synced through `sql` parent.
 
 # Usage
 
+Add script in your `package.json`
+```json
+{
+  "name": "my-package",
+  "scripts": {
+    "sync:db": "sync-db",
```
Comment (mesaugat) [inline README.md]: `:` is our convention. It is better to use the same format.

```json
"sync-db": "sync-db"
```

File: README.md [hunk 26]
```diff
@@ -99,16 +99,28 @@ sql:
...
+  }
+}
+```
+
+Run
+
 ```bash
-$ npx sync-db
+$ yarn sync:db
```
Comment (mesaugat) [inline README.md]: sync-db

**Pull request merged**